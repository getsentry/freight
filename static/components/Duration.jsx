var React = require("react");

var Duration = React.createClass({
  propTypes: {
    seconds: React.PropTypes.number.isRequired
  },

  getDuration() {
    var result, neg;
    var value = this.props.seconds * 1000;

    neg = value < 0 ? true : false;
    if (neg) {
      value = -value;
    }

    if (value > 7200000) {
      result = Math.round(value / 3600000) + ' hr';
    } else if (value > 120000) {
      result = Math.round(value / 60000) + ' min';
    } else if (value > 10000) {
      result = Math.round(value / 1000) + ' sec';
    } else if (value > 1000) {
      result = Math.round(value / 1000) + ' sec';
    } else {
      result = Math.round(value) + ' ms';
    }

    if (neg) {
      result = '-' + result;
    }

    return result;
  },

  render() {
    return (
      <span className={this.props.className}>{this.getDuration()}</span>
    );
  }
});

export default Duration;
