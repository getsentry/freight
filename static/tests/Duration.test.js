import React from 'react';
import {mount} from 'enzyme';
import Duration from '../components/Duration.jsx';

test('Duration Snapshot', () => {
  const wrapper = mount(<Duration seconds={13.67} className="duration" />);
  expect(wrapper).toMatchSnapshot();
});
