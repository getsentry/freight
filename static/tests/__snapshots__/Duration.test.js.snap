// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duration Snapshot 1`] = `
<Duration
  className="duration"
  seconds={13.67}
>
  <span
    className="duration"
  >
    14 sec
  </span>
</Duration>
`;
